cluster/store.go

```diff
@@ -641,3 +647,82 @@ func lastSnapshotIndex(ss *raft.FileSnapshotStore) uint64 {
 	}
 	return ls[0].Index
 }
+
+// recoverSingleNode is used to manually force a new configuration in order to
+// recover from a loss of quorum where the current configuration cannot be
+// WARNING! This operation implicitly commits all entries in the Raft log, so
+// in general this is an extremely unsafe operation and that's why it's made to be
+// used in a single cluster node.
+// for more details see : https://github.com/hashicorp/raft/blob/main/api.go#L279
+func (st *Store) recoverSingleNode() error {
+	if st.cfg.BootstrapExpect > 1 || len(st.candidates) > 1 {
+		return fmt.Errorf("bootstrap expect  %v, candidates %v,"+
```

Comment:
NIT: one too many space between `expect  %v` and no space when concatenating too string with `+` which will lead to a weird error message.